var/spack/repos/builtin.mock/packages/depb/package.py

```diff
@@ -6,7 +6,7 @@
 from spack.package import *
 
 
-class Depb(AutotoolsPackage):
+class Depb(Package):
```

Comment:
This changes seems fine but a couple other edits to this package are needed:

* The docstring is no longer accurate (inheriting `AutotoolsPackage` adds a build-only dependency on `autotools`); technically `depends_on(b)` adds a build dependency (since the default dependency type is build/link), but I think the comment will be confusing unless reworded to something like "simple package with one dependency")
* The comment in the `install()` method also needs to be updated

While all the tests pass, I'm curious if you did a pass over unit tests that:

* Install `depb` directly or
* a parent of it

and if they seem to expect that `depb` is autotools based (this is mainly to make sure that unit tests that were originally testing something don't regress).

Reply:
Okay, I will make the additional edits. As for checking the unit tests, I did see if installing depb directly in other test files successfully passed with the change. This was indeed the case (Ex. concretize.py). However, I didn't thoroughly look into tests that used the parent of depb which is b (now know as pkg-b with the new changes made 3 days ago by Haampie). Though, what I did check passed.

Reply:
There appears be one test that claims to make use of (or refer to) `depb` having a build dependency in `test_env_install_two_specs_same_dep`/`test/cmd/env.py`

```
    """Test installation of two packages that share a dependency with no
    connection and the second specifying the dependency as a 'build'
    dependency.
    """
```

Given the test name and overall behavior, I don't *think* it was essential for the packages to share a build-only dep, but I'm curious if @tldahlgren you agree.

If @tldahlgren agrees that wasn't necessary, can you likewise edit the docstring for that test @nicolecheetham?

(all other direct references looked good - accounting for the change you made in `test/cmd/list` which also looks good - and nothing currently depends on `depb`).

Reply:
I've gone back over the history and have to agree that the `build` type aspect isn't what is critical.